common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1130,24 +1142,36 @@ public static String[] getICULocaleNames() {
 
   public static UTF8String getCollationKey(UTF8String input, int collationId) {
     Collation collation = fetchCollation(collationId);
+    Collation.CollationSpec.SpaceTrimming spaceTrimming =
+      Collation.CollationSpec.getSpaceTrimming(collationId);
+    if (spaceTrimming != Collation.CollationSpec.SpaceTrimming.NONE) {
```

Comment:
since we're likely to be performing these checks all over the place, I'd say this looks like something that would best fit in a utility method (within CollationFactory for example)

Reply:
It's needed in full form, the reason is you fetch spaceTrimming that you need to applyTrimmingCollation.

Reply:
that's okay, just implement it differently

Reply:
please see `public static boolean isCaseSensitiveAndAccentInsensitive` and consider adding something like `public static boolean isTrimSensitive`

Reply:
Sorry, don't see any value in doing so, further why call another function?
So it would be something like public static boolean isTrimSensitive(int collationId), but I already needed to extract spaceTrimming enum from collation id, so perform 2 redudant operations?

Reply:
`public static boolean isTrimSensitive(SpaceTrimming spaceTrimming)`?

Reply:
i don't see it being different from inline spaceTrimming != SpaceTrimming.NONE.

Reply:
of course the functionality is the same, but can we focus on:
- avoiding code repetition
- proper logic centralization

please investigate for ways to achieve this on your own - if the above proposal doesn't work for you, please see `public final boolean supportsLowercaseEquality;` and consider adding something like `public final boolean supportsTrimming;` to the `Collation` class

then we can simply have:
```
Collation collation = fetchCollation(collationId);
if (collation.supportsTrimming) ...
```